client/dashboard/me/security-account-recovery/recovery-email.tsx

```diff
@@ -0,0 +1,224 @@
+import {
+	profileQuery,
+	accountRecoveryQuery,
+	updateAccountRecoveryEmailMutation,
+	removeAccountRecoveryEmailMutation,
+	resendAccountRecoveryEmailValidationMutation,
+} from '@automattic/api-queries';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import {
+	__experimentalHStack as HStack,
+	__experimentalInputControl as InputControl,
+	__experimentalVStack as VStack,
+	__experimentalSpacer as Spacer,
+	__experimentalConfirmDialog as ConfirmDialog,
+	Button,
+	Card,
+	CardBody,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { DataForm } from '@wordpress/dataviews';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useMemo, useState, useEffect } from 'react';
+import Notice from '../../components/notice';
+import { SectionHeader } from '../../components/section-header';
+import type { Field } from '@wordpress/dataviews';
+
+type SecurityEmailFormData = {
+	email: string;
+};
+
+export default function RecoveryEmail() {
+	const [ formData, setFormData ] = useState< SecurityEmailFormData >( {
+		email: '',
+	} );
+	const [ isRemoveDialogOpen, setIsRemoveDialogOpen ] = useState( false );
+	const [ showResendButton, setShowResendButton ] = useState( true );
+
+	const { data: accountRecoveryData, isLoading: isAccountRecoveryDataLoading } = useQuery(
+		accountRecoveryQuery()
+	);
+
+	const { mutate: validateEmailMutation, isPending: isValidateEmailPending } = useMutation(
+		updateAccountRecoveryEmailMutation()
+	);
+
+	const { mutate: removeEmailMutation, isPending: isRemoveEmailPending } = useMutation(
+		removeAccountRecoveryEmailMutation()
+	);
+
+	const { mutate: resendValidationMutation, isPending: isResendValidationPending } = useMutation(
+		resendAccountRecoveryEmailValidationMutation()
+	);
+
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+
+	const accountRecoveryEmail = accountRecoveryData?.email;
+
+	useEffect( () => {
```

Comment:
This can cause "blinking form" when the data is not ready yet for the first time.

I think a better way is to:

1. Preload the data in the router to ensure the data is ready in the component. See e.g.:

https://github.com/Automattic/wp-calypso/blob/88188c49c638a9aed87096e2c193478aaebc3d23/client/dashboard/app/router/sites.tsx#L375

2. Use `useSuspenseQuery()` instead of `useQuery()` to get the `accountRecoveryData`.

3. We can simply initialize the `formData` using the above, e.g.

```ts
const [ formData, setFormData ] = useState< SecurityEmailFormData >( {
	accountRecoveryData.email: '',
} );
```

Reply:
Oh, I didn't know we had something like this. This is way neater appraoach. I have updated it here: https://github.com/Automattic/wp-calypso/pull/105504/commits/1255f0357fb13037ded38e60c4c4b52473d40e97